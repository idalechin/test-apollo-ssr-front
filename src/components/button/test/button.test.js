import React from "react";
import { shallow, mount } from "enzyme";
import {render, fireEvent, cleanup} from 'react-testing-library'
import Button from "../index";
import renderer from "react-test-renderer";
import testIcon from "./ic_star_black_24px.svg";
import activeTestIcon from "./ic_star_border_black_24px.svg";

afterEach(cleanup)

const btnText = "Check button";
const className = "btn-test";
const mockCallback = jest.fn();

describe('<Button />', () => {
  it("Test renders right", () => {
    const btn = renderer
      .create(
        <Button
          text={btnText}
          icon={testIcon}
          className={className}
          activeIcon={activeTestIcon}
          disable
          toggle
          onClick={() => console.log(btnText)}
        />
      )
      .toJSON();
    expect(btn).toMatchSnapshot();
  });

  it("Test click", () => {
    const button = shallow(<Button onClick={mockCallback} />);
    button.find('button').simulate('click');
    expect(mockCallback.mock.calls.length).toEqual(1);
  });

  it("Test active click", () => {
    const {container} = render(<Button onClick={mockCallback} toggle={true}/>)
    const btn = container.firstChild

    expect(btn.classList.contains('active')).toBe(false)

    fireEvent.click(btn)

    expect(btn.classList.contains('active')).toBe(true)
  });
})
