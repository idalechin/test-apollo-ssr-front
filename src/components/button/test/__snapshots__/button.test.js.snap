// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Test renders right 1`] = `
<button
  className="btn btn-test  disabled"
  onClick={[Function]}
>
  <svg
    className="btn__icon"
    viewBox="undefined"
  >
    <use
      xlinkHref="#undefined"
    />
  </svg>
  <span
    className="btn__text"
  >
    Check button
  </span>
</button>
`;
